### PR TITLE
feat: added lots of new documentation for getting started and using the design system

### LINKED ISSUE #487: Add "past versions" page

### PR DESCRIPTION
**Pull request by daylesalmon**
This PR is heavily focused on brand new documentation that assist with getting started and using the design system.

There is a brand new directory called docs which uses a new layout template (similar to left-sidebar).

Fixes #487

### CODE REVIEW

**Review by mrmjprice — DISMISSED** [review]

File: src/www/_layouts/layout-docs.njk [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    <li>
+      <a href="/user-research/" class="wmnds-link">User research</a>
+    </li>
+    <li>
+      <a href="/docs/releases/" class="wmnds-link">Releases</a>
+    </li>
+  </ul>
+  </div>
+  <div class="wmnds-col-1 wmnds-col-lg-2-3" id="wmnds-main-content">
+    {# Show section as smmall text above title #}
```
Comment (catiarodriguescosta) [inline src/www/_layouts/layout-docs.njk:55]:
```suggestion
    {# Show section as small text above title #}
```

File: src/www/_layouts/layout-left-pane.njk [hunk 2]
```diff
@@ -1,11 +1,17 @@
 {% extends "www/_layouts/_master-layout.njk" %}
+{% set section = section if section else "Components" %}
 
 {% block body %}
 <div class="wmnds-grid wmnds-grid--spacing-lg-2-lg">
   <div class="wmnds-col-1 wmnds-col-lg-1-3 wmnds-p-b-lg">
     {% include "www/_partials/nav/subnav.njk" %}
   </div>
   <div class="wmnds-col-1 wmnds-col-lg-2-3" id="wmnds-main-content">
+    {# Show section as smmall text above title #}
```
Comment (catiarodriguescosta) [inline src/www/_layouts/layout-left-pane.njk:10]:
```suggestion
    {# Show section as small text above title #}
```

File: src/www/pages/docs/get-started/design/index.njk.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+{% markdown %}
+
+## About
+
+We use <a href="https://www.sketch.com/" target="_blank" rel="noopener noreferrer">Sketch</a> to create and prototype designs for our [styles](/styles/), [components](/components/) and [patterns](/patterns/) within the West Midlands Network Design System.
+
+## Sketch library
+
+If you would like to create designs based on what we have already done, you can <a href="https://forms.office.com/Pages/ResponsePage.aspx?id=RetZCK7xCk6e-ubWa7tnL0kEZK0X_-9IoNQ__PZJI49UNlBZUFRPNENVTFRWV08xQk1SN0FPR0dDQi4u" target='_blank'>request the Sketch library</a>.
```
Comment (catiarodriguescosta) [inline src/www/pages/docs/get-started/design/index.njk.md:16]:
```suggestion
If you would like to create designs based on what we have already done, you can <a href="https://forms.office.com/Pages/ResponsePage.aspx?id=RetZCK7xCk6e-ubWa7tnL0kEZK0X_-9IoNQ__PZJI49UNlBZUFRPNENVTFRWV08xQk1SN0FPR0dDQi4u" target='_blank' rel='noopener noreferrer'>request the Sketch library</a>.
```

File: src/www/pages/docs/get-started/index.njk.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+The guide, [using the design system](using-the-design-system/) contains more information on how to use the West Midlands Network Design System.
+
+---
+
+## Before making a service live
+
+You must:
+
+- Make sure you've tested your service with real users
+- Make sure you meet <a href="https://www.gov.uk/guidance/make-your-website-or-app-accessible-and-publish-an-accessibility-statement?utm_source=CampaignPage1&utm_campaign=access_regs" target='_blank'>accessibility regulations</a> else you might be <a href="https://www.legislation.gov.uk/uksi/2018/952/made" title="The Public Sector Bodies (Websites and Mobile Applications) (No. 2) Accessibility Regulations 2018" target='_blank'>breaking the law</a>
```
Comment (catiarodriguescosta) [inline src/www/pages/docs/get-started/index.njk.md:34]:
```suggestion
- Make sure you meet <a href="https://www.gov.uk/guidance/make-your-website-or-app-accessible-and-publish-an-accessibility-statement?utm_source=CampaignPage1&utm_campaign=access_regs" target='_blank' rel='noopener noreferrer'>accessibility regulations</a> else you might be <a href="https://www.legislation.gov.uk/uksi/2018/952/made" title="The Public Sector Bodies (Websites and Mobile Applications) (No. 2) Accessibility Regulations 2018" target='_blank' rel='noopener noreferrer'>breaking the law</a>
```

File: src/www/pages/docs/get-started/index.njk.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+- <a href="https://forms.office.com/Pages/ResponsePage.aspx?id=RetZCK7xCk6e-ubWa7tnL0kEZK0X_-9IoNQ__PZJI49UNlBZUFRPNENVTFRWV08xQk1SN0FPR0dDQi4u" title="Let us know you're using the design system" target='_blank'>Let us know</a> you're using the design system on a live service
```
Comment (catiarodriguescosta) [inline src/www/pages/docs/get-started/index.njk.md:35]:
```suggestion
- <a href="https://forms.office.com/Pages/ResponsePage.aspx?id=RetZCK7xCk6e-ubWa7tnL0kEZK0X_-9IoNQ__PZJI49UNlBZUFRPNENVTFRWV08xQk1SN0FPR0dDQi4u" title="Let us know you're using the design system" target='_blank' rel='noopener noreferrer'>Let us know</a> you're using the design system on a live service
```

File: src/www/pages/docs/get-started/production/index.njk.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+{% block content %}
+
+{% markdown %}
+
+## About
+
+To start using our [styles](/styles/), [components](/components/) and [patterns](/patterns/) in your production project, you'll first need to include West Midlands Network Design System in your project.
+
+There are currently two ways of including the design system within your poject. You can either install it using node package manager (npm) or include the compiled files in your application.
```
Comment (catiarodriguescosta) [inline src/www/pages/docs/get-started/production/index.njk.md:14]:
```suggestion
There are currently two ways of including the design system within your project. You can either install it using node package manager (npm) or include the compiled files in your application.
```

File: src/www/pages/docs/releases/index.njk.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+Using this method of versioning enables West Midlands Network Design System to release new versions without breaking any projects dependant on previous releases.
+
+This method also allows you to upgrade your project to the latest version of the design system when you are able to.
+
+---
+
+## Accessing a previous releases
+
+It is recommened that you use the latest release of West Midlands Network Design System in your project. This guide explains how to access a previous release for older documentation.
```
Comment (catiarodriguescosta) [inline src/www/pages/docs/releases/index.njk.md:22]:
```suggestion
It is recommended that you use the latest release of West Midlands Network Design System in your project. This guide explains how to access a previous release for older documentation.
```

File: src/www/pages/index.njk [hunk 8]
```diff
@@ -91,60 +104,15 @@
...
-<br>
-<h3>Get help with using the design system</h3>
-<p>The design system is maintained by the Digital and Data team at West Midlands Combined Authority.</p>
-<p>You can <a href="https://forms.office.com/Pages/ResponsePage.aspx?id=RetZCK7xCk6e-ubWa7tnL0kEZK0X_-9IoNQ__PZJI49UNlBZUFRPNENVTFRWV08xQk1SN0FPR0dDQi4u" title="Ask a question about using the design system" target='_blank'>ask a question using this form</a>.</p>
+
+{# Get help #}
+<div class="wmnds-col-1 wmnds-col-md-3-4">
+  <h2>Get help with using the design system</h2>
+  <p>The design system is maintained by the Digital and Data team at West Midlands Combined Authority.</p>
+  <p>You can <a href="https://forms.office.com/Pages/ResponsePage.aspx?id=RetZCK7xCk6e-ubWa7tnL0kEZK0X_-9IoNQ__PZJI49UNlBZUFRPNENVTFRWV08xQk1SN0FPR0dDQi4u" title="Ask a question about using the design system" target='_blank'>ask a question using this form</a>.</p>
```
Comment (catiarodriguescosta) [inline src/www/pages/index.njk:115]:
```suggestion
  <p>You can <a href="https://forms.office.com/Pages/ResponsePage.aspx?id=RetZCK7xCk6e-ubWa7tnL0kEZK0X_-9IoNQ__PZJI49UNlBZUFRPNENVTFRWV08xQk1SN0FPR0dDQi4u" title="Ask a question about using the design system" target='_blank' rel='noopener noreferrer'>ask a question using this form</a>.</p>
```

File: src/www/pages/docs/get-started/production/index.njk.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+         <link rel="stylesheet" href="https://unpkg.com/wmn-design-system@$*version/build/css/wmnds.min.css" />
+       </head>
+       <body>
+         <!-- site content... -->
+       </body>
+     </html>
+     {%- endfilter %}
+     </code></pre>
+
+2. Add <a href='/styles/icons/#using-icons' title='Documentation about using icons' target='_blank'>West Midlands Network Design System icons</a> to your project.
```
Comment (catiarodriguescosta) [inline src/www/pages/docs/get-started/production/index.njk.md:40]:
```suggestion
2. Add <a href='/styles/icons/#using-icons' title='Documentation about using icons' target='_blank' rel='noopener noreferrer'>West Midlands Network Design System icons</a> to your project.
```

File: src/www/_layouts/layout-docs.njk [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+      <a href="/docs/get-started/using-the-design-system/" class="wmnds-link">Using the design system</a>
+    </li>
+  </ul>
+
+  <hr class="wmnds-col-1 wmnds-col-sm-3-4"/>
+
+  <h4>Other</h4>
+  <ul class="wmnds-website-subnav">
+    <li>
+      <a href="/user-research/" class="wmnds-link">User research</a>
```
Comment (catiarodriguescosta) [inline src/www/_layouts/layout-docs.njk:47]: Having this page opening in a different template bugged me a little. 
If we can't change the template maybe we could change the link to open in a new tab...

**Review by catiarodriguescosta — CHANGES_REQUESTED** [review]
Great work here @daylesalmon! 👍 
Just left a few tiny amends as suggestions.

**Review by jkmnch — CHANGES_REQUESTED** [review]
On this page: https://deploy-preview-590--wmnetwork.netlify.app/docs/get-started/using-the-design-system/

The line: "When using Nunjucks macros in the Prototype Kit leave out the first line that starts with" ends suddenly. Is this right?

Also, in the second paragraph of the Javascript section, can you replace 'don't' with do not as we're not supposed to use negative contractions in copy.

File: src/www/pages/docs/get-started/production/index.njk.md (hunk 9 above)
Comment (daylesalmon) [reply src/www/pages/docs/get-started/production/index.njk.md:40]: Not going to add the rel="noopener noreferrer" to this one as it is our website, so we want to promote SEO link growth.

**Comment by daylesalmon** [comment]
> On this page: https://deploy-preview-590--wmnetwork.netlify.app/docs/get-started/using-the-design-system/
> 
> The line: "When using Nunjucks macros in the Prototype Kit leave out the first line that starts with" ends suddenly. Is this right?
> 
> Also, in the second paragraph of the Javascript section, can you replace 'don't' with 'do not' as we're not supposed to use negative contractions in copy.

This has now been resolved. Good spot!

**Review by jkmnch — APPROVED** [review]

**Review by catiarodriguescosta — APPROVED** [review]
Looking good! 👍

**Pull request merged**